From 7fd1d69b1aaedd0ce451e356941ab7e8c6f17977 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:59 -0400
Subject: [usb] xhci: Check for XHCI_PLAT in xhci_cleanup_msix()

Message-id: <1410887463-20674-228-git-send-email-dzickus@redhat.com>
Patchwork-id: 95670
O-Subject: [RHEL7 PATCH 227/591] usb: xhci: Check for XHCI_PLAT in xhci_cleanup_msix()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9005355af23856c55a5538c9024355785424821b
Author: Jack Pham <jackp@codeaurora.org>
Date:   Fri Nov 15 14:53:14 2013 -0800

    usb: xhci: Check for XHCI_PLAT in xhci_cleanup_msix()

    If CONFIG_PCI is enabled, make sure xhci_cleanup_msix()
    doesn't try to free a bogus PCI IRQ or dereference an invalid
    pci_dev when the xHCI device is actually a platform_device.

    This patch should be backported to kernels as old as 3.9, that
    contain the commit 52fb61250a7a132b0cfb9f4a1060a1f3c49e5a25
    "xhci-plat: Don't enable legacy PCI interrupts."

    Signed-off-by: Jack Pham <jackp@codeaurora.org>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Cc: stable@vger.kernel.org
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 6bc966c..f8ffc51 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -325,6 +325,9 @@ static void xhci_cleanup_msix(struct xhci_hcd *xhci)
  struct usb_hcd *hcd = xhci_to_hcd(xhci);
  struct pci_dev *pdev = to_pci_dev(hcd->self.controller);
 
+ if (xhci->quirks & XHCI_PLAT)
+  return;
+
  xhci_free_irq(xhci);
 
  if (xhci->msix_entries) {
-- 
1.7.1